### PR TITLE
Feat/replaced yargs with envalid

### PR DESCRIPTION
**Pull request by Ellenn-A**
# Pull Request

## Checklist
- [x] Have you read Digital Catapult's [Code of Conduct](https://github.com/digicatapult/.github/blob/main/CODE_OF_CONDUCT.md)?
- [x] I have performed a self-review of my own code.
- [ ] I have commented my code, particularly in hard-to-understand areas.
- [x] I have made corresponding changes to the documentation.
- [x] My changes generate no new warnings.
- [ ] I have added tests that prove my fix is effective or that my feature works.
- [ ] New and existing unit tests pass locally with my changes.

## PR Type

Please delete options that are irrelevant.

- [ ] Bug Fix
- [x] Feature
- [x] Documentation Update
- [ ] Code style update (formatting, local variables)
- [ ] Breaking Change (fix or feature that would cause existing functionality to change)

## Linked tickets

- https://digicatapult.atlassian.net/browse/VR-122

## High level description

Replace yargs with envalid, as we do not need to provide CLI arguments and can deal entirely using ENVARs

## Detailed description

Replaced yargs with envalid, removed CLI options, updated (where possible) docker-compose files to now use envars

## Describe alternatives you've considered

N/A

## Operational impact
Allows for variables to be set as Envs.

## Additional context

I don't really know what to do about the depcheck error because I am importing dotenv like so: `import dotenv from 'dotenv'` right at the top of `src>env.ts`

### CODE REVIEW

File: README_VERITABLE.md [hunk 1]
```diff
@@ -43,21 +43,21 @@ The project aims to enable supply chains to share insights and data across multi
...
 | Argument added | Required | Default | Description |
 | :------------- | :------: | :------ | :------- |
-| ipfs-origin | Y | http://localhost:5001 | IPFS endpoint |
-|opa-origin |N |http://localhost:8181 |OPA endpoint |
-|persona-title | N |"Veritable Cloudagent"|Tab name which you can see in your browser |
-|persona-color |N |"white" |Defines the background colour of swagger documentation|
+| IPFS_ORIGIN | Y | http://localhost:5001 | IPFS endpoint |
+|OPA_ORIGIN |N |http://localhost:8181 |OPA endpoint |
+|PERSONA_TITLE | N |"Veritable Cloudagent"|Tab name which you can see in your browser |
+|PERSONA_COLOR |N |"white" |Defines the background colour of swagger documentation|
```
Comment (mattdean-digicatapult) [inline README_VERITABLE.md:60]: We need to document the envs better here I think

File: src/env.ts [hunk 2]
```diff
@@ -0,0 +1,111 @@
...
+
+const envConfig = {
+  LABEL: envalid.str({ default: 'AFJ Rest', devDefault: 'AFJ Rest Agent' }),
+  WALLET_ID: envalid.str({ default: 'walletId', devDefault: 'walletId' }),
+  WALLET_KEY: envalid.str({ default: 'walletKey', devDefault: 'walletKey' }),
+  ENDPOINT: stringArray({
+    default: ['http://localhost:5002', 'ws://localhost:5003'],
+    devDefault: ['http://localhost:5002', 'ws://localhost:5003'],
+  }),
+  LOG_LEVEL: envalid.num({ default: 3, devDefault: 3 }),
```
Comment (mattdean-digicatapult) [inline src/env.ts:48]: I thought we'd replaced the logger with `pino`. Should this not be choices like `trace`, `debug`, `info` etc.?

File: src/env.ts [hunk 3]
```diff
@@ -0,0 +1,111 @@
+import dotenv from 'dotenv'
+import * as envalid from 'envalid'
+import { makeValidator } from 'envalid'
+import { singleton } from 'tsyringe'
+
+if (process.env.NODE_ENV === 'test') {
+  dotenv.config({ path: 'tests/test.env' }) //different configs for unit and integration??
```
Comment (mattdean-digicatapult) [inline src/env.ts:7]: does this file exist? If not it might be worth creating it just so we can set `LOG_LEVEL=silent`

**Review by mattdean-digicatapult — CHANGES_REQUESTED** [review]
Main thing is the README where we need to explain options available. There is a question on the logger config that might not be best done here, but I'd like to understand what's going on

File: README_VERITABLE.md (hunk 1 above)
Comment (Ellenn-A) [reply README_VERITABLE.md:60]: would you want all envs documented?

Comment (mattdean-digicatapult) [reply README_VERITABLE.md:60]: yes please

File: README_VERITABLE.md [hunk 4]
```diff
@@ -43,21 +43,62 @@ The project aims to enable supply chains to share insights and data across multi
...
-|persona-color |N |"white" |Defines the background colour of swagger documentation|
+| LABEL| Y|"AFJ Rest"|Label for the agent|
+|WALLET_ID| Y| "walletId"| An id of the Agent's wallet|
+|WALLET_KEY| Y|"walletKey"| A key for the Agent's wallet|
+|ENDPOINT|Y|['http://localhost:5002', 'ws://localhost:5003']| An array of endpoint for the agent app|
+|LOG_LEVEL|Y| info| Log level for the app |
+|USE_DID_SOV_PREFIX_WHERE_ALLOWED|N|false|Allows the usage of 'sov' prefix in DIDs where possible|
+|USE_DID_KEY_IN_PROTOCOLS|N|true| Allows the use of DID keys in protocols|
+|OUTBOUND_TRANSPORT|Y|['http', 'ws']|Specifies the type of outbound transport|
+|INBOUND_TRANSPORT|Y|"[{"transport": "http", "port": 5002}, {"transport": "ws", "port": 5003}]"|Specifies the inbound transport, needs to be provided as a JSON parseable string|
```
Comment (mattdean-digicatapult) [inline README_VERITABLE.md:65]: I actually prefer this as json exactly like you've done. Can you check though with @dblane-digicatapult that this isn't too horrible from an infra PoV

File: src/env.ts [hunk 5]
```diff
@@ -1,10 +1,11 @@
-import dotenv from 'dotenv'
+import { LogLevel } from '@credo-ts/core'
+import * as dotenv from 'dotenv'
 import * as envalid from 'envalid'
 import { makeValidator } from 'envalid'
 import { singleton } from 'tsyringe'
 
 if (process.env.NODE_ENV === 'test') {
-  dotenv.config({ path: 'tests/test.env' }) //different configs for unit and integration??
+  dotenv.config({ path: 'tests/test.env.ts' })
```
Comment (mattdean-digicatapult) [inline src/env.ts:8]: Sorry this seems to have been lost in translation. Just create the `test/test.env` file with the contents

```
LOG_LEVEL=3
```

File: src/env.ts [hunk 6]
```diff
@@ -0,0 +1,112 @@
...
+  }
+  try {
+    return input.split(',').map((s) => s.trim())
+  } catch (err) {
+    throw new Error('Invalid input for string array')
+  }
+})
+
+const envConfig = {
+  LABEL: envalid.str({ default: 'AFJ Rest', devDefault: 'AFJ Rest Agent' }),
```
Comment (dblane-digicatapult) [inline src/env.ts:42]: This should be replaced with veritable now?

Also what is this used for?

Comment (mattdean-digicatapult) [reply src/env.ts:42]: It appears in all the back and forth didcom messages and is important

Comment (Ellenn-A) [reply src/env.ts:42]: updated the docs

**Review by mattdean-digicatapult — APPROVED** [review]

**Pull request merged**